src/linear_algebra/gram_schmidt_ortho.lean

```diff
@@ -0,0 +1,259 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao, Kevin Buzzard
+-/
+
+import tactic.basic
+import algebra.big_operators.basic
+import analysis.inner_product_space.basic
+import analysis.inner_product_space.projection
+import analysis.normed_space.is_R_or_C
+
+/-!
+# Gram-Schmidt Orthogonalization and Orthonormalization
+
+In this file we introduce Gram-Schmidt Orthogonalization and Orthonormalization.
+
+The Gram-Schmidt process takes a set of vectors as input
+and outputs a set of orthogonal vectors which have the same span.
+
+## Main results
+
+- `gram_schmidt_process` : Gram-Schmidt process
+- `gram_schmidt_process_orthogonal` :
+  the proof that "gram_schmidt_process" produces an orthogonal system of vectors
+- `gram_schmidt_process_normed` :
+  Normalized "Gram-Schmidt" (i.e each vector in this system has unit length)
+- `gram_schmidt_process_orthornormal` :
+  the proof that "gram_schmidt_process_normed" produces an orthornormal system of vectors
+- `gram_schmidt_process_span_eq` :
+  the proof that `gram_schmidt_process` preserves the span of vectors
+-/
+
+open_locale big_operators
+
+variables (𝕜 : Type*) (E : Type*) [is_R_or_C 𝕜] [inner_product_space 𝕜 E]
+
+/-- Gram-Schmidt process -/
+noncomputable def gram_schmidt_process (f : ℕ → E) : ℕ → E
+| n := f n - ∑ i : fin n, have ↑i < n := i.prop,
+  orthogonal_projection (𝕜 ∙ gram_schmidt_process i) (f n)
+
+/-- 'gram_schmidt_process_def' turns the sum over `fin n` into a sum over `ℕ`. -/
+lemma gram_schmidt_process_def (f : ℕ → E) (n : ℕ) :
+  gram_schmidt_process 𝕜 E f n = f n - ∑ i in finset.range n,
+    orthogonal_projection (𝕜 ∙ gram_schmidt_process 𝕜 E f i) (f n) :=
+begin
+  rw gram_schmidt_process,
+  congr' 1,
+  exact fin.sum_univ_eq_sum_range (λ i,
+    (orthogonal_projection (𝕜 ∙ gram_schmidt_process 𝕜 E f i) (f n) : E)) n,
+end
+
+@[simp] lemma gram_schmidt_process_zero (f : ℕ → E) :
+  gram_schmidt_process 𝕜 E f 0 = f 0 := by simp [gram_schmidt_process]
+
+/-- **Gram-Schmidt Orthogonalisation**
+Gram-Schmidt process produces an orthogonal system of vectors. -/
+theorem gram_schmidt_process_orthogonal (f : ℕ → E) (a b : ℕ) (h₀ : a < b) :
+  (inner (gram_schmidt_process 𝕜 E f a) (gram_schmidt_process 𝕜 E f b) : 𝕜) = 0 :=
```

Comment:
```suggestion
Gram-Schmidt process produces an orthogonal system of vectors. -/
theorem gram_schmidt_process_orthogonal (f : ℕ → E) (a b : ℕ) (h₀ : a < b) :
  ⟪gram_schmidt_process 𝕜 E f a, gram_schmidt_process 𝕜 E f b⟫ = 0 :=
```
For this notation to work, you would need to add
```
local notation `⟪`x`, `y`⟫` := @inner 𝕜 _ _ x y
```
at the top of the file just after the definition of `𝕜`. (This is what is done in most files involving inner products, see e.g. `inner_product_space/projection.lean`.

Reply:
oh yes, I forgot to use notation here which is clearer. Thank you! I have changed it